src/Microsoft.Framework.PackageManager/Publish/PublishManager.cs

```diff
@@ -112,7 +113,19 @@ public bool Publish()
 
             foreach (var runtime in _options.Runtimes)
             {
-                var frameworkName = DependencyContext.GetFrameworkNameForRuntime(Path.GetFileName(runtime));
+                var frameworkName = DependencyContext.SelectFrameworkNameForRuntime(
+                    project.GetTargetFrameworks().Select(x => x.FrameworkName), 
+                    Path.GetFileName(runtime));
+
+                if (frameworkName == null)
+                {
+                    _options.Reports.Error.WriteLine(
+                        string.Format("The project being published does not support the runtime '{0}'",
```

Comment:
We don't need `string.Format` here, `IReport.WriteLine` supports formatted output.